### PR TITLE
Rework of PRs #10025 and #10046 -  Rules and Decoders for ESET Remote Administrator

### LINKED ISSUE #10970: Rework of Issues 9991 & 9742 - Rules and Decoders for ESET Remote Administrator
|Wazuh version|Component|Install type|Install method|Platform|Action type|
|---|---|---|---|---|---|
| 4.3.0 | Rules and Decoders | Manager | Packages | Any | Rework|

|Related issues|Related PR|
|---|---|
| #9742 | #10025  |
| #9991 | #10046 |

## Description
This issue is has been created as a rework of the rules and decoders added in issue [#9991](https://github.com/wazuh/wazuh/issues/9991) and [#9742](https://github.com/wazuh/wazuh/issues/9742). Issue [#9742](https://github.com/wazuh/wazuh/issues/9742)  covers the community requested use case of adding rules and decoders for ESET Remote Administrator, while [#9991](https://github.com/wazuh/wazuh/issues/9991) is an extension to [#9742](https://github.com/wazuh/wazuh/issues/9742) modifying the log type for ESET Protect. 
The aim of this issue is:
- To ensure the rules and decoders from issue [#9991](https://github.com/wazuh/wazuh/issues/9991) and [#9742](https://github.com/wazuh/wazuh/issues/9742) work correctly.
- To fix issues regarding indentation, syntax, rule granularity, spellings and ensure the ruleset quality is standard.

## Original Issue
Type: Community Request

### Request from #9742

I need you to analyze these decoders and rules and consider adding them to the ruleset of Wazuh.
**Decoders:**
```
<decoder name="eset-main">
  <prematch>^\<\d+>\d+ \d+-\d+-\d+T\d+:\d+:\d+.\d+\w \w+ \w+ \d+</prematch>
</decoder>

<decoder name="eset-main-child">
  <parent>eset-main</parent>
  <prematch>-\s+</prematch>
  <plugin_decoder offset="after_prematch">JSON_Decoder</plugin_decoder>
</decoder>

<decoder name="eset-main-child">
  <parent>eset-main</parent>
  <regex>^\<\d+>\d+ (\d+-\d+-\d+T\d+:\d+:\d+.\d+\w) (\w+) (\w+) (\d+)</regex>
  <order>timestamp,type,source,msg_id</order>
</decoder>
```

**Rules:**
```
<group name="eset,">

    <rule id="770000" level="0">
        <decoded_as>eset-main</decoded_as>
        <description>Eset Console Logs</description>
    </rule>

    <rule id="770001" level="5">
        <if_sid>770000</if_sid>
        <field name="severity">^Warning$</field>
        <description>Eset: Warning message was logged from $(hostname)</description>
    </rule>

    <rule id="770002" level="7">
        <if_sid>770000</if_sid>
        <field name="severity">^Error$</field>
        <description>Eset: Error message was logged from $(hostname)</description>
    </rule>

</group>
```
[Slack Thread](https://wazuh.slack.com/archives/C0A933R8E/p1629224285064400?thread_ts=1629157947.005700&cid=C0A933R8E) with some log Samples

I hope this information could be helpful.

### Request from #9991

Consider adding the decoders and rules wrote and tested for ESET Protect Console.

**Decoders**
I based my decoders on the following Log Sample:
```
May  6 10:59:37 XXXXX ERAServer[5032]: {"event_type":"Audit_Event","ipv4":"XXX.XXX.XXX.XXX","hostname":"XXXXX","source_uuid":"9416183d-3XX3-4776-9783-9532a3a027bb","occured":"06-May-2021 09:59:37","severity":"Information","domain":"Domain group","action":"Login attempt","target":"a49d257e-ecc6-4063-95c6-5eb5e6b3e5df","detail":"Authenticating domain user 'XXXXXXXX'.","user":"","result":"Success"}
```
This log is in a BSD format, so to achieve this log decoding, it is needed to configure in the ESET Protect Console, the Syslog output in BSD format. Please look at the following reference to configure it: https://help.eset.com/era_admin/65/en-US/admin_server_settings_syslog.html

Once this is done, and the Wazuh Manager is able to collect Syslog messages from that source, we need to set the following decoders in the `/var/ossec/etc/decoders` folder, I created the file named `eset-bsd-decoders.xml`:
```
<decoder name="eset-bsd">
  <program_name>ERAServer</program_name>
</decoder>

<decoder name="eset-main-child">
  <parent>eset-bsd</parent>
  <plugin_decoder offset="after_parent">JSON_Decoder</plugin_decoder>
</decoder>
```

**Rules**
For the rules, I relied on the following article: https://help.eset.com/protect_admin/81/en-US/events-exported-to-json-format.html
It shows some events and their main parts and the different type of logs. The rules I wrote are these:
```
<!-- Reference: https://help.eset.com/protect_admin/81/en-US/events-exported-to-json-format.html -->
<group name="eset,">

    <rule id="770000" level="0">
        <decoded_as>eset-bsd</decoded_as>
        <description>Eset Console Logs</description>
    </rule>

    <rule id="770001" level="3">
        <if_sid>770000</if_sid>
        <field name="event_type">^Threat_Event$</field>
        <description>Eset: Threat Event rules Group</description>
        <group>threat_event,</group>
    </rule>

    <rule id="770002" level="3">
        <if_sid>770000</if_sid>
        <field name="event_type">^FirewallAggregated_Event$</field>
        <description>Eset: Firewall Aggregated rules Group</description>
        <group>firewallaggregated_event,</group>
    </rule>

    <rule id="770003" level="3">
        <if_sid>770000</if_sid>
        <field name="event_type">^HipsAggregated_Event$</field>
        <description>Eset: HIPS Aggregated rules Group</description>
        <group>hipsaggregated_event,</group>
    </rule>

    <rule id="770004" level="2">
        <if_sid>770000</if_sid>
        <field name="event_type">^Audit_Event$</field>
        <description>Eset: Audit rules Group</description>
        <group>audit_event,</group>
    </rule>

    <rule id="770005" level="3">
        <if_sid>770000</if_sid>
        <field name="event_type">^EnterpriseInspectorAlert_Event$</field>
        <description>Eset: Enterprise Inspector Alert rules Group</description>
        <group>enterpriseinspectoralert_event,</group>
    </rule>

    <rule id="770006" level="3">
        <if_sid>770000</if_sid>
        <field name="event_type">^EnterpriseInspectorAlert_Event$</field>
        <description>Eset: Enterprise Inspector Alert rules Group</description>
        <group>enterpriseinspectoralert_event,</group>
    </rule>

    <rule id="770010" level="5">
        <if_sid>770000</if_sid>
        <field name="severity">^Warning$</field>
        <description>Eset: Warning message was logged from $(hostname)</description>
    </rule>

    <rule id="770011" level="7">
        <if_sid>770000</if_sid>
        <field name="severity">^Error$</field>
        <description>Eset: Error message was logged from $(hostname)</description>
    </rule>

    <rule id="770012" level="9">
        <if_sid>770000</if_sid>
        <field name="severity">^Critical$</field>
        <description>Eset: Critical message was logged from $(hostname)</description>
    </rule>

</group>
```

With these rules and decoders, Wazuh will be able to fully support ESET Protect Console events natively.
I hope this could be helpful.

### PR DESCRIPTION
**Pull request by xrisbarney**
|Related issue|
|---|
|#10970 |

## Description

This PR aims to resolve issues detected under #10025 and #10046 PRs. Closes #10970 
The issues resolved are:
- Missing copyright headers.
- Missing software details.
- Fixed indentation issues.
- Updated the rule levels.
- Added the rule ID range.

### Syntax

- [x] 1.a Rule tags order must be compliant.
- [x] 1.b Decoder tags order must be compliant.
- [x] 1.c XML blocks must as compact as possible.
- [X] 1.d Only one empty line between rule/decoder and the next rule/decoder.
- [X] 1.e Decoder extracted fields names must use '_' whether a space is needed.
- [X] 1.f Decoder name must use '-' whether a space is needed.
 
### Grammar

- [X] 2.a Grammar quality.
- [X] 2.b Similar phrases must keep tenses and expressions.
- [X] 2.c Grammar basic rules like capitalization, punctuation marks, sentence construction, and others.

### Semantic

- [x] 3.a New decoder, rule, or test are written in the correct file and grouped correctly inside the file.
- [x] 3.b Group similar rules under same ID's group.
- [x] 3.c: 
     - [x] 3.c1 Find and reuse group name before creating a new one.
     - [x] 3.c2 Include a new group definition in a PR comment.
     - [x] 3.c3 Any group name must use '_' whether it does need a space char.
     - [x] 3.c4 The groups inside a tag must be sorted in alphabetical order.
- [x] 3.d Rule level should be compliant with the [documentation](https://documentation.wazuh.com/current/user-manual/ruleset/rules-classification.html).

### Unit testing

- [X] 4.a A metadata block at the beginning of the .ini file describing software, version, and logs source.
- [X] 4.b Each new rule must have at least one test entry in the correct .ini file.
- [x] 4.c Runtest.py must pass and must include the results in raw format here.
Test of eset.ini

```
root@ubuntu-focal:/var/ossec/ruleset/feed/ruleset/testing# python3 runtests.py -t tests/eset.ini 
Restarting wazuh-manager...
- [ File = ./tests/eset.ini ] ---------
........

|Component |  Tested  |  Total   | Coverage |
| -------- | -------- | -------- | -------- |
|  Rules   |    8     |   4174   |  0.19%   |
| Decoders |    1     |   172    |  0.58%   |

|          File           |  Passed  |  Failed  |  Status  |
|        --------         | -------- | -------- | -------- |
|./tests/eset.ini         |    8     |    0     |    ✅     |
Restarting wazuh-manager...
```

<details><summary>All rules and decoders test.</summary>
<p>

```
root@ubuntu-focal:/var/ossec/ruleset/feed/ruleset/testing# python3 runtests.py
Restarting wazuh-manager...
- [ File = ./tests/samba.ini ] ---------
....

- [ File = ./tests/fortigate.ini ] ---------
........................................

- [ File = ./tests/apparmor.ini ] ---------
.....

- [ File = ./tests/pam.ini ] ---------
.....

- [ File = ./tests/audit_lateral.ini ] ---------
......

- [ File = ./tests/systemd.ini ] ---------
..

- [ File = ./tests/arbor.ini ] ---------
..

- [ File = ./tests/dovecot.ini ] ---------
...............

- [ File = ./tests/mailscanner.ini ] ---------
.

- [ File = ./tests/ossec.ini ] ---------
.....

- [ File = ./tests/vuln_detector.ini ] ---------
..

- [ File = ./tests/squid_rules.ini ] ---------
..

- [ File = ./tests/pix.ini ] ---------
......................

- [ File = ./tests/huawei_usg.ini ] ---------
...

- [ File = ./tests/checkpoint_smart1.ini ] ---------
..................

- [ File = ./tests/junos.ini ] ---------
...

- [ File = ./tests/web_appsec.ini ] ---------
...............................

- [ File = ./tests/cimserver.ini ] ---------
..

- [ File = ./tests/cisco_ftd.ini ] ---------
..........................................

- [ File = ./tests/firewalld.ini ] ---------
..

- [ File = ./tests/audit_recon.ini ] ---------
..

- [ File = ./tests/openldap.ini ] ---------
.........

- [ File = ./tests/sophos_fw.ini ] ---------
..........

- [ File = ./tests/owlh.ini ] ---------
....

- [ File = ./tests/freepbx.ini ] ---------
......

- [ File = ./tests/syslog.ini ] ---------
......

- [ File = ./tests/openvpn_ldap.ini ] ---------
..

- [ File = ./tests/sophos.ini ] ---------
........

- [ File = ./tests/unbound.ini ] ---------


- [ File = ./tests/github.ini ] ---------
....................................................................................................................................................................................................................................................................................................................................

- [ File = ./tests/cpanel.ini ] ---------
.......

- [ File = ./tests/fireeye.ini ] ---------
...

- [ File = ./tests/fortiauth.ini ] ---------
....

- [ File = ./tests/iptables.ini ] ---------
........

- [ File = ./tests/named.ini ] ---------
.....

- [ File = ./tests/sophos-utm-firewall.ini ] ---------
......

- [ File = ./tests/nginx.ini ] ---------
............

- [ File = ./tests/exim.ini ] ---------
.......

- [ File = ./tests/opensmtpd.ini ] ---------
.......

- [ File = ./tests/oscap.ini ] ---------
................................

- [ File = ./tests/gitlab.ini ] ---------
.........................

- [ File = ./tests/paloalto.ini ] ---------
................

- [ File = ./tests/api.ini ] ---------
............................

- [ File = ./tests/aws_s3_access.ini ] ---------
.......

- [ File = ./tests/exchange.ini ] ---------
..

- [ File = ./tests/f5_big_ip.ini ] ---------
...........................................

- [ File = ./tests/cisco_asa.ini ] ---------
........................................................................................

- [ File = ./tests/nextcloud.ini ] ---------
.......

- [ File = ./tests/trendmicro-cloud-one.ini ] ---------
.................

- [ File = ./tests/pfsense.ini ] ---------
..

- [ File = ./tests/vsftpd.ini ] ---------
....

- [ File = ./tests/web_rules.ini ] ---------
..........

- [ File = ./tests/apache.ini ] ---------
..................

- [ File = ./tests/eset.ini ] ---------
........

- [ File = ./tests/doas.ini ] ---------
....

- [ File = ./tests/sysmon.ini ] ---------
...

- [ File = ./tests/gcp.ini ] ---------
...........

- [ File = ./tests/icinga.ini ] ---------
....

- [ File = ./tests/postfix.ini ] ---------
..

- [ File = ./tests/audit_scp.ini ] ---------
.

- [ File = ./tests/dropbear.ini ] ---------
...

- [ File = ./tests/rsh.ini ] ---------
..

- [ File = ./tests/kernel_usb.ini ] ---------
......

- [ File = ./tests/mcafee_epo.ini ] ---------
.

- [ File = ./tests/netscreen.ini ] ---------
....

- [ File = ./tests/panda_paps.ini ] ---------
........

- [ File = ./tests/auditd.ini ] ---------
...

- [ File = ./tests/cylance.ini ] ---------
.......

- [ File = ./tests/modsecurity.ini ] ---------
......

- [ File = ./tests/cloudlfare-waf.ini ] ---------
.............

- [ File = ./tests/office365.ini ] ---------
................................................................................................................................

- [ File = ./tests/sshd.ini ] ---------
...........................................

- [ File = ./tests/cisco_ios.ini ] ---------
.................

- [ File = ./tests/glpi.ini ] ---------
...

- [ File = ./tests/sudo.ini ] ---------
........

- [ File = ./tests/su.ini ] ---------
.....

- [ File = ./tests/proftpd.ini ] ---------
.......

- [ File = ./tests/SonicWall.ini ] ---------
........

|Component |  Tested  |  Total   | Coverage |
| -------- | -------- | -------- | -------- |
|  Rules   |   982    |   4174   |  23.53%  |
| Decoders |    76    |   172    |  44.19%  |

|          File           |  Passed  |  Failed  |  Status  |
|        --------         | -------- | -------- | -------- |
|./tests/samba.ini        |    4     |    0     |    ✅     |
|./tests/fortigate.ini    |    40    |    0     |    ✅     |
|./tests/apparmor.ini     |    5     |    0     |    ✅     |
|./tests/pam.ini          |    5     |    0     |    ✅     |
|./tests/audit_lateral.ini|    6     |    0     |    ✅     |
|./tests/systemd.ini      |    2     |    0     |    ✅     |
|./tests/arbor.ini        |    2     |    0     |    ✅     |
|./tests/dovecot.ini      |    15    |    0     |    ✅     |
|./tests/mailscanner.ini  |    1     |    0     |    ✅     |
|./tests/ossec.ini        |    5     |    0     |    ✅     |
|./tests/vuln_detector.ini|    2     |    0     |    ✅     |
|./tests/squid_rules.ini  |    2     |    0     |    ✅     |
|./tests/pix.ini          |    22    |    0     |    ✅     |
|./tests/huawei_usg.ini   |    3     |    0     |    ✅     |
|./tests/checkpoint_smart1.ini|    18    |    0     |    ✅     |
|./tests/junos.ini        |    3     |    0     |    ✅     |
|./tests/web_appsec.ini   |    31    |    0     |    ✅     |
|./tests/cimserver.ini    |    2     |    0     |    ✅     |
|./tests/cisco_ftd.ini    |    42    |    0     |    ✅     |
|./tests/firewalld.ini    |    2     |    0     |    ✅     |
|./tests/audit_recon.ini  |    2     |    0     |    ✅     |
|./tests/openldap.ini     |    9     |    0     |    ✅     |
|./tests/sophos_fw.ini    |    10    |    0     |    ✅     |
|./tests/owlh.ini         |    4     |    0     |    ✅     |
|./tests/freepbx.ini      |    6     |    0     |    ✅     |
|./tests/syslog.ini       |    6     |    0     |    ✅     |
|./tests/openvpn_ldap.ini |    2     |    0     |    ✅     |
|./tests/sophos.ini       |    8     |    0     |    ✅     |
|./tests/unbound.ini      |    0     |    0     |    ✅     |
|./tests/github.ini       |   324    |    0     |    ✅     |
|./tests/cpanel.ini       |    7     |    0     |    ✅     |
|./tests/fireeye.ini      |    3     |    0     |    ✅     |
|./tests/fortiauth.ini    |    4     |    0     |    ✅     |
|./tests/iptables.ini     |    8     |    0     |    ✅     |
|./tests/named.ini        |    5     |    0     |    ✅     |
|./tests/sophos-utm-firewall.ini|    6     |    0     |    ✅     |
|./tests/nginx.ini        |    12    |    0     |    ✅     |
|./tests/exim.ini         |    7     |    0     |    ✅     |
|./tests/opensmtpd.ini    |    7     |    0     |    ✅     |
|./tests/oscap.ini        |    32    |    0     |    ✅     |
|./tests/gitlab.ini       |    25    |    0     |    ✅     |
|./tests/paloalto.ini     |    16    |    0     |    ✅     |
|./tests/api.ini          |    28    |    0     |    ✅     |
|./tests/aws_s3_access.ini|    7     |    0     |    ✅     |
|./tests/exchange.ini     |    2     |    0     |    ✅     |
|./tests/f5_big_ip.ini    |    43    |    0     |    ✅     |
|./tests/cisco_asa.ini    |    88    |    0     |    ✅     |
|./tests/nextcloud.ini    |    7     |    0     |    ✅     |
|./tests/trendmicro-cloud-one.ini|    17    |    0     |    ✅     |
|./tests/pfsense.ini      |    2     |    0     |    ✅     |
|./tests/vsftpd.ini       |    4     |    0     |    ✅     |
|./tests/web_rules.ini    |    10    |    0     |    ✅     |
|./tests/apache.ini       |    18    |    0     |    ✅     |
|./tests/eset.ini         |    8     |    0     |    ✅     |
|./tests/doas.ini         |    4     |    0     |    ✅     |
|./tests/sysmon.ini       |    3     |    0     |    ✅     |
|./tests/gcp.ini          |    11    |    0     |    ✅     |
|./tests/icinga.ini       |    4     |    0     |    ✅     |
|./tests/postfix.ini      |    2     |    0     |    ✅     |
|./tests/audit_scp.ini    |    1     |    0     |    ✅     |
|./tests/dropbear.ini     |    3     |    0     |    ✅     |
|./tests/rsh.ini          |    2     |    0     |    ✅     |
|./tests/kernel_usb.ini   |    6     |    0     |    ✅     |
|./tests/mcafee_epo.ini   |    1     |    0     |    ✅     |
|./tests/netscreen.ini    |    4     |    0     |    ✅     |
|./tests/panda_paps.ini   |    8     |    0     |    ✅     |
|./tests/auditd.ini       |    3     |    0     |    ✅     |
|./tests/cylance.ini      |    7     |    0     |    ✅     |
|./tests/modsecurity.ini  |    6     |    0     |    ✅     |
|./tests/cloudlfare-waf.ini|    13    |    0     |    ✅     |
|./tests/office365.ini    |   128    |    0     |    ✅     |
|./tests/sshd.ini         |    43    |    0     |    ✅     |
|./tests/cisco_ios.ini    |    17    |    0     |    ✅     |
|./tests/glpi.ini         |    3     |    0     |    ✅     |
|./tests/sudo.ini         |    8     |    0     |    ✅     |
|./tests/su.ini           |    5     |    0     |    ✅     |
|./tests/proftpd.ini      |    7     |    0     |    ✅     |
|./tests/SonicWall.ini    |    8     |    0     |    ✅     |
Restarting wazuh-manager...
```
</p>
</details>

- [X] 4.d New CDB lists must include the proper test entry in the correct .ini file.

### E2E testing

- [x]5.a Logs for new or modified decoder/rules sent to a manager running with this PR ruleset appear in Kibana.
image

- [x]5.b There are not affected or broken visualizations on Kibana.
image
image

- [x]5.c New or modified items can be seen correctly using APP ruleset navigation.
image

### Elasticsearch Template

- [X] 6.a Known fields with output format managed and usually used for searching are included in template array index.query.default_field. 
- [X] 6.b The new field with the correct date format is stored as a "date" type field in the template.
- [X] 6.c The new extracted IP fields are in the pipeline as "geo" field and "geo_point" type in the template. 
- [X] 6.d Known fields with output format managed and usually used for searching are included in the template.

### Stoppers

- [x] 7.a No previous rule ID changes without triple check.
- [x] 7.b No previous decoder name changes without triple check.
- [x] 7.c No previous file name changes without triple check.
- [x] 7.d No previous test changes its 'rule' field value without triple check.

### Others

- [x] 8.a Each file has the correct copyright block.
- [x] 8.b The copyright block doesn't have "Author" only "Created by Wazuh". To include an "Author" request triple check.
- [x] 8.c The copyright block doesn't use "-". 
- [x] 8.d The rule files don't have any sample log.
- [x] 8.e The decoder file has sample logs next to the decoder that matches that log.
- [x] 8.f The decoder and rule files have information about software, version, and any helpful information.
- [x] 8.g The PR has a single commit with CHANGELOG changes in the correct format.
- [x] 8.h The new rule ID is not in use.
- [x] 8.i The new rule ID must be in the defined IDs range. 
- [x] 8.j New rules ID range must be verified with a triple check and noted in the rules ID document.
- [x] 8.k The new extracted IP fields are in the pipeline as "geo" field and "geo_point" type in the template.

### CODE REVIEW

File: ruleset/decoders/0575-eset-remote_decoders.xml [hunk 1]
```diff
@@ -1,6 +1,27 @@
-<!-- Sample logs
-May  6 10:59:37 XXXXX ERAServer[5032]: {"event_type":"Audit_Event","ipv4":"XXX.XXX.XXX.XXX","hostname":"XXXXX","source_uuid":"9416183d-3XX3-4776-9783-9532a3a027bb","occured":"06-May-2021 09:59:37","severity":"Information","domain":"Domain group","action":"Login attempt","target":"a49d257e-ecc6-4063-95c6-5eb5e6b3e5df","detail":"Authenticating domain user 'XXXXXXXX'.","user":"","result":"Success"}
+<!--
+  ESET Remote decoders
+  Created by Wazuh, Inc.
+  Copyright (C) 2015-2021, Wazuh Inc.
+  This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2.
```
Comment (jcruzlp) [inline ruleset/decoders/0575-eset-remote_decoders.xml:5]: New copyright block is this one:
```xml
<!--
  Copyright (C) 2015-2021, Wazuh Inc.
-->
```

File: ruleset/decoders/0575-eset-remote_decoders.xml [hunk 2]
```diff
@@ -1,6 +1,27 @@
...
+    Reference: https://help.eset.com/era_admin/65/en-US/admin_server_settings_syslog.html
+  
+  Software details: 
+  - This decoder was created using logs from the software with the following details:
+    Software Name: ESET Remote Administrator
+    Software Version: 8.1
+
+  Sample logs:
+  May  6 10:59:37 XXXXX ERAServer[5032]: {"event_type":"Audit_Event","ipv4":"XXX.XXX.XXX.XXX","hostname":"XXXXX","source_uuid":"9416183d-3XX3-4776-9783-9532a3a027bb","occured":"06-May-2021 09:59:37","severity":"Information","domain":"Domain group","action":"Login attempt","target":"a49d257e-ecc6-4063-95c6-5eb5e6b3e5df","detail":"Authenticating domain user 'XXXXXXXX'.","user":"","result":"Success"}
+-->
```
Comment (jcruzlp) [inline ruleset/decoders/0575-eset-remote_decoders.xml:23]: Comment's can't have `-`, check 8c

File: ruleset/rules/0925-eset-remote_rules.xml [hunk 3]
```diff
@@ -1,62 +1,83 @@
...
+  - The reference below shows some events, their main parts and their different log types.
+    Reference: https://help.eset.com/protect_admin/81/en-US/events-exported-to-json-format.html
+  
+  Software details: 
+  - These rules were created using logs from the software with the following details:
+    Software Name: ESET Remote Administrator
+    Software Version: 8.1
+  
+  ID range: 42001 - 42100
+-->
```
Comment (jcruzlp) [inline ruleset/rules/0925-eset-remote_rules.xml:21]: Ditto

File: ruleset/rules/0925-eset-remote_rules.xml [hunk 4]
```diff
@@ -1,62 +1,83 @@
...
+
 <group name="eset,">
 
   <rule id="42001" level="0">
-      <decoded_as>eset-bsd</decoded_as>
-      <description>Eset Console Logs</description>
+    <decoded_as>eset-bsd</decoded_as>
+    <description>Eset console logs.</description>
```
Comment (jcruzlp) [inline ruleset/rules/0925-eset-remote_rules.xml:27]: It should be ESET: <info>

**Review by jcruzlp — CHANGES_REQUESTED** [review]

File: ruleset/rules/0925-eset-remote_rules.xml (hunk 4 above)
Comment (xrisbarney) [reply ruleset/rules/0925-eset-remote_rules.xml]: Fixed, thanks.

File: ruleset/rules/0925-eset-remote_rules.xml (hunk 3 above)
Comment (xrisbarney) [reply ruleset/rules/0925-eset-remote_rules.xml]: Resolved.

File: ruleset/decoders/0575-eset-remote_decoders.xml [hunk 5]
```diff
@@ -1,6 +1,24 @@
...
+    These decoders accepts logs generated by the ESET Remote Administrator and decodes them.
+  
+    The logs from ESET are generated in BSD format, so to achieve log decoding,
+    in the ESET Protect Console Syslog output format has to be configured.
+  
+    The supporting reference for configuring the ESET logs to output as Syslog is below.
+    Reference: https://help.eset.com/era_admin/65/en-US/admin_server_settings_syslog.html
+
+  Sample logs:
+  May  6 10:59:37 XXXXX ERAServer[5032]: {"event_type":"Audit_Event","ipv4":"XXX.XXX.XXX.XXX","hostname":"XXXXX","source_uuid":"9416183d-3XX3-4776-9783-9532a3a027bb","occured":"06-May-2021 09:59:37","severity":"Information","domain":"Domain group","action":"Login attempt","target":"a49d257e-ecc6-4063-95c6-5eb5e6b3e5df","detail":"Authenticating domain user 'XXXXXXXX'.","user":"","result":"Success"}
```
Comment (72nomada) [inline ruleset/decoders/0575-eset-remote_decoders.xml:19]: indent

File: ruleset/decoders/0575-eset-remote_decoders.xml [hunk 6]
```diff
@@ -1,6 +1,24 @@
...
+-->
+
 <decoder name="eset-bsd">
   <program_name>ERAServer</program_name>
```
Comment (72nomada) [inline ruleset/decoders/0575-eset-remote_decoders.xml:23]: make program_name restricted as possible: 

`^ERAServer$`

**Review by 72nomada — CHANGES_REQUESTED** [review]
In parent decoder force to ensure only ESETSerer is consider

**Review by jcruzlp — APPROVED** [review]
LGTM!

**Review by 72nomada — APPROVED** [review]
LGTM

**Comment by Spades0** [comment]
LGTM

File: ruleset/decoders/0575-eset-remote_decoders.xml [hunk 7]
```diff
@@ -1,8 +1,26 @@
...
+    The supporting reference for configuring the ESET logs to output as Syslog is below.
+    Reference: https://help.eset.com/era_admin/65/en-US/admin_server_settings_syslog.html
+
+  Sample logs:
+    May  6 10:59:37 XXXXX ERAServer[5032]: {"event_type":"Audit_Event","ipv4":"XXX.XXX.XXX.XXX","hostname":"XXXXX","source_uuid":"9416183d-3XX3-4776-9783-9532a3a027bb","occured":"06-May-2021 09:59:37","severity":"Information","domain":"Domain group","action":"Login attempt","target":"a49d257e-ecc6-4063-95c6-5eb5e6b3e5df","detail":"Authenticating domain user 'XXXXXXXX'.","user":"","result":"Success"}
+-->
+
 <decoder name="eset-bsd">
-  <program_name>ERAServer</program_name>
+  <program_name>^ERAServer$</program_name>
```
Comment (fabamatic) [inline ruleset/decoders/0575-eset-remote_decoders.xml:23]: Are you sure about this change?

Comment (fabamatic) [reply ruleset/decoders/0575-eset-remote_decoders.xml:23]: I mean, no version number or anything like that can come appended to program name?

**Review by fabamatic — APPROVED** [review]
LGTM

**Comment by JcabreraC** [comment]
# Test output
`# python2 runtests.py `
```bash

- [ File = ./tests/apparmor.ini ] ---------
.....

- [ File = ./tests/ossec.ini ] ---------
.....

- [ File = ./tests/mailscanner.ini ] ---------
.

- [ File = ./tests/vsftpd.ini ] ---------
....

- [ File = ./tests/gitlab.ini ] ---------
.........................

- [ File = ./tests/aws_s3_access.ini ] ---------
.......

- [ File = ./tests/opensmtpd.ini ] ---------
.......

- [ File = ./tests/paloalto.ini ] ---------
................

- [ File = ./tests/nextcloud.ini ] ---------
.......

- [ File = ./tests/nginx.ini ] ---------
............

- [ File = ./tests/netscreen.ini ] ---------
....

- [ File = ./tests/auditd.ini ] ---------
...

- [ File = ./tests/cisco_ftd.ini ] ---------
..........................................

- [ File = ./tests/kernel_usb.ini ] ---------
......

- [ File = ./tests/openldap.ini ] ---------
.........

- [ File = ./tests/huawei_usg.ini ] ---------
...

- [ File = ./tests/api.ini ] ---------
............................

- [ File = ./tests/doas.ini ] ---------
....

- [ File = ./tests/cpanel.ini ] ---------
.......

- [ File = ./tests/office365.ini ] ---------
................................................................................................................................

- [ File = ./tests/postfix.ini ] ---------
..

- [ File = ./tests/firewalld.ini ] ---------
..

- [ File = ./tests/rsh.ini ] ---------
..

- [ File = ./tests/exchange.ini ] ---------
..

- [ File = ./tests/sudo.ini ] ---------
........

- [ File = ./tests/arbor.ini ] ---------
..

- [ File = ./tests/sophos-utm-firewall.ini ] ---------
......

- [ File = ./tests/icinga.ini ] ---------
....

- [ File = ./tests/exim.ini ] ---------
.......

- [ File = ./tests/pam.ini ] ---------
.....

- [ File = ./tests/named.ini ] ---------
.....

- [ File = ./tests/oscap.ini ] ---------
................................

- [ File = ./tests/SonicWall.ini ] ---------
........

- [ File = ./tests/proftpd.ini ] ---------
.......

- [ File = ./tests/sysmon.ini ] ---------
...

- [ File = ./tests/fortiauth.ini ] ---------
....

- [ File = ./tests/modsecurity.ini ] ---------
......

- [ File = ./tests/f5_big_ip.ini ] ---------
...........................................

- [ File = ./tests/cimserver.ini ] ---------
..

- [ File = ./tests/iptables.ini ] ---------
........

- [ File = ./tests/syslog.ini ] ---------
......

- [ File = ./tests/junos.ini ] ---------
...

- [ File = ./tests/cloudlfare-waf.ini ] ---------
.............

- [ File = ./tests/trendmicro-cloud-one.ini ] ---------
.................

- [ File = ./tests/mcafee_epo.ini ] ---------
.

- [ File = ./tests/web_appsec.ini ] ---------
...............................

- [ File = ./tests/sophos.ini ] ---------
........

- [ File = ./tests/dovecot.ini ] ---------
...............

- [ File = ./tests/web_rules.ini ] ---------
..........

- [ File = ./tests/github.ini ] ---------
....................................................................................................................................................................................................................................................................................................................................

- [ File = ./tests/owlh.ini ] ---------
....

- [ File = ./tests/cisco_asa.ini ] ---------
........................................................................................

- [ File = ./tests/su.ini ] ---------
.....

- [ File = ./tests/gcp.ini ] ---------
...........

- [ File = ./tests/sshd.ini ] ---------
...........................................

- [ File = ./tests/panda_paps.ini ] ---------
........

- [ File = ./tests/freepbx.ini ] ---------
......

- [ File = ./tests/fireeye.ini ] ---------
...

- [ File = ./tests/cisco_ios.ini ] ---------
.................

- [ File = ./tests/pfsense.ini ] ---------
..

- [ File = ./tests/cylance.ini ] ---------
.......

- [ File = ./tests/dropbear.ini ] ---------
...

- [ File = ./tests/pix.ini ] ---------
......................

- [ File = ./tests/squid_rules.ini ] ---------
..

- [ File = ./tests/samba.ini ] ---------
....

- [ File = ./tests/sophos_fw.ini ] ---------
..........

- [ File = ./tests/audit_recon.ini ] ---------
..

- [ File = ./tests/audit_scp.ini ] ---------
.

- [ File = ./tests/glpi.ini ] ---------
...

- [ File = ./tests/apache.ini ] ---------
..................

- [ File = ./tests/eset.ini ] ---------
........

- [ File = ./tests/checkpoint_smart1.ini ] ---------
..................

- [ File = ./tests/fortigate.ini ] ---------
........................................

- [ File = ./tests/unbound.ini ] ---------


- [ File = ./tests/audit_lateral.ini ] ---------
......

- [ File = ./tests/openvpn_ldap.ini ] ---------
..

- [ File = ./tests/systemd.ini ] ---------
..

- [ File = ./tests/vuln_detector.ini ] ---------
..
```

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**